bayes_search.py

```diff
@@ -145,51 +165,75 @@ def train_gaussian_process(
     return gp, y_mean, y_stddev
 
 
-def filter_weird_values(sample_X, sample_y):
+def filter_nans(sample_X: npt.ArrayLike, sample_y: npt.ArrayLike) -> npt.ArrayLike:
     is_row_finite = ~(np.isnan(sample_X).any(axis=1) | np.isnan(sample_y))
     sample_X = sample_X[is_row_finite, :]
     sample_y = sample_y[is_row_finite]
     return sample_X, sample_y
 
 
 def next_sample(
-    sample_X,
-    sample_y,
-    X_bounds=None,
-    runtimes=None,
-    failures=None,
-    current_X=None,
-    nu=1.5,
-    max_samples_for_gp=100,
-    improvement=0.01,
-    num_points_to_try=1000,
-    opt_func="expected_improvement",
-    test_X=None,
-):
+    sample_X: npt.ArrayLike,
```

Comment:
since this is python3.8 what do you think about keyword only arguments for this function since it is so complex

Reply:
done.